Tools/gdb/libpython.py

```diff
@@ -1392,7 +1392,7 @@ def write_repr(self, out, visited):
 
 
 def int_from_int(gdbval):
-    return int(str(gdbval))
+    return int(str(gdbval), 0)
```

Comment:
Can one not simply call `int` on integer-type GDB values?

Reply:
You mean something like

```suggestion
    return int(gdbval.cast(gdb.lookup_type('long')))
```
?

Reply:
Yes, but do you have to do the `cast`?

Reply:
I think so - but to be sure I have to verify it. Will update my branch if it works.

Reply:
I've updated my branch. Seems a simple `int` call is enough - thanks (I assume this only holds if the gdbvalue has the right type).